api/datastore/sql/migrations/migrations.go

```diff
@@ -126,6 +155,11 @@ func DownAll(driver string, db *sql.DB) error {
 func ApplyMigrations(ctx context.Context, driver string, db *sql.DB) error {
 	log := common.Logger(ctx)
 
+	// no point to handle error from here, basically err
+	// indicates about any problem with old migrations table
+	current, _, err := checkOldMigrationTableVersionIfExists(db)
```

Comment:
if the dirty bit is set, we need to exit the program and suggest the user to fix their db manually (I don't expect this to happen, but i'd rather have this than not and try to debug it)

Reply:
fixed, failing if dirty